src/test/ui/macros/rfc-3086-metavar-expr/dollar-dollar-has-correct-behavior.rs

```diff
@@ -0,0 +1,38 @@
+// run-pass
+
+#![feature(macro_metavar_expr)]
+
+macro_rules! nested {
+    ( $a:ident ) => {
+        macro_rules! $a {
+            ( $$( $b:ident ),* ) => {
+                $$(
+                    macro_rules! $b {
+                        ( $$$$( $c:ident ),* ) => {
+                            $$$$(
+                                fn $c() -> &'static str { stringify!($c) }
+                            ),*
+                        };
+                    }
+                )*
+            };
+        }
+    };
+}
+
+macro_rules! not_nested {
+    ( $$ $foo:ident ) => {{
```

Comment:
https://github.com/rust-lang/rust/pull/93545#discussion_r814837555 is still relevant, both `$$` and `${...}` on LHS should be tested as errors.

Reply:
Can I please resolve this scenario in a following PR? The following should also be placed together in the same lot. 

* https://github.com/rust-lang/rust/pull/94368/files#diff-75de1e3838473080bc68d366c767ae46e6e3bdcf302cd9525093113f9025c945R327-R329
* https://github.com/rust-lang/rust/pull/94368/files#diff-75de1e3838473080bc68d366c767ae46e6e3bdcf302cd9525093113f9025c945R442-R444
*  https://github.com/rust-lang/rust/pull/94368/files#diff-75de1e3838473080bc68d366c767ae46e6e3bdcf302cd9525093113f9025c945R442-R444

Reply:
The fix I expect (modulo adding the test cases) is simply using `if parsing_patterns` one more time for `$$`.

Reply:
Oh... Indeed!

Just added the fix along side more tests